prov/rxm/src/rxm_cq.c

```diff
@@ -1357,6 +1357,13 @@ void rxm_ep_do_progress(struct util_ep *util_ep)
 		rxm_conn_process_eq_events(rxm_ep);
 
 	while (!rxm_rx_buf_acquire(rxm_ep, &buf)) {
+		// Discard rx buffer if its msg_ep was
+		// closed after eq processing.
+		if (!buf->conn->msg_ep) {
+			rxm_rx_buf_release(rxm_ep, buf, 0);
+			continue;
+		}
```

Comment:
This looks like the wrong place to have this cleanup.  We're in the progress function.  I would have expected this cleanup to occur closer to the msg ep being closed, where conn->msg_ep is set to null.

Reply:
@shefty I agree that it not the best place in code to release buffer.
My first solution to solve the problem was quite similar to you proposal. But to use it in addition to the main rxm_ep repost list we should track all buffers binded to each particular connection. This solution adds some unnecessary complexity, don't you think so?

Reply:
If we maintain the RX buffer pool per individual connection, we can do the cleanup when the MSG EP is closed. In fact, there won't be a need to release the buffers individually. We can just call `ofi_bufpool_destroy` as long as we're are using `OFI_BUFPOOL_NO_TRACK` when creating the pool. I believe there shouldn't be any change in memory consumption since we already fill up the receive queue for every MSG endpoint that we open.

Reply:
If there's an rx buffer pool per domain shared among the msg eps, is cleanup needed at all?  I'm guessing the only per msg ep data being recorded is which msg ep each rx buffer has been posted to.  Is this right?  If so, we could just set that when posting the buffer.

Reply:
The rx buffer pool is per rxm_ep, so there can be multiple buffer pools per MSG provider domain.

> I'm guessing the only per msg ep data being recorded is which msg ep each rx buffer has been posted to. Is this right? If so, we could just set that when posting the buffer

I'm not sure if I follow this. If we get an rx buffer from the pool for a MSG EP that's been closed, what do we do with that? We can't replace the closed MSG EP (rx_buf->msg_ep) with some other MSG EP as we don't know which one to pick (unless we maintain per MSG EP counters).

Reply:
My comment was based on sharing a single rx buffer pool among multiple msg eps  I can't tell from the comments if this is the case or not.

Reply:
> My comment was based on sharing a single rx buffer pool among multiple msg eps I can't tell from the comments if this is the case or not.

This is the case in the current code (master).